### PR TITLE
Add support for retry jobs

### PR DESCRIPTION
**Pull request by shabtaisharon**

### CODE REVIEW

File: SpectraLogic.SpectraStorageBrokerClient/ResponseParsers/RetryResponseParser.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using log4net;
+using SpectraLogic.SpectraStorageBrokerClient.Model;
+using SpectraLogic.SpectraStorageBrokerClient.Runtime;
+using System.Net;
+
+namespace SpectraLogic.SpectraStorageBrokerClient.ResponseParsers
+{
+    internal class RetryResponseParser : IResponseParser<IJob>
+    {
+        #region Fields
```
Comment (RachelTucker) [inline SpectraLogic.SpectraStorageBrokerClient/ResponseParsers/RetryResponseParser.cs]: any particular reason your using the `#region` here?

Comment (shabtaisharon) [reply SpectraLogic.SpectraStorageBrokerClient/ResponseParsers/RetryResponseParser.cs]: This region wraps the Fields in this class

Comment (RachelTucker) [reply SpectraLogic.SpectraStorageBrokerClient/ResponseParsers/RetryResponseParser.cs]: Why? What does wrapping the fields achieve?

Comment (shabtaisharon) [reply SpectraLogic.SpectraStorageBrokerClient/ResponseParsers/RetryResponseParser.cs]: it's just a design decision in this project

Comment (RachelTucker) [reply SpectraLogic.SpectraStorageBrokerClient/ResponseParsers/RetryResponseParser.cs]: fair enough

Comment (rpmoore) [reply SpectraLogic.SpectraStorageBrokerClient/ResponseParsers/RetryResponseParser.cs]: It allows visual studio to collapse the region down.

File: SpectraLogic.SpectraStorageBrokerClient/Model/JobStatusEnum.cs [hunk 2]
```diff
@@ -26,24 +25,29 @@ public enum JobStatusEnum
...
         /// <summary>
-        /// The job is paused
+        /// The job has error
         /// </summary>
-        PAUSED,
+        ERROR,
 
         /// <summary>
-        /// The job is canceled
+        /// The job is paused
```
Comment (rpmoore) [inline SpectraLogic.SpectraStorageBrokerClient/Model/JobStatusEnum.cs]: PAUSED should never be returned as a status.

File: SpectraLogic.SpectraStorageBrokerClient/Model/JobStatusEnum.cs [hunk 3]
```diff
@@ -26,24 +25,29 @@ public enum JobStatusEnum
...
         /// </summary>
-        CANCELED,
+        PAUSED,
 
         /// <summary>
-        /// The job has error
+        /// The job is retrying
         /// </summary>
-        ERROR
+        RETRY
```
Comment (rpmoore) [inline SpectraLogic.SpectraStorageBrokerClient/Model/JobStatusEnum.cs]: RETRY is not a job state other, but an action that results in a new job being created.

File: SpectraLogic.SpectraStorageBrokerClient/Model/JobType.cs [hunk 4]
```diff
@@ -26,18 +26,23 @@ public enum JobType
...
-        /// The cancel
+        /// The restore
+        /// </summary>
+        RESTORE,
+
+        /// <summary>
+        /// The retry
         /// </summary>
-        CANCEL
+        RETRY
```
Comment (rpmoore) [inline SpectraLogic.SpectraStorageBrokerClient/Model/JobType.cs]: This is not a job type.

**Comment by shabtaisharon** [comment]
@rpmoore if you can look again at the changes and merge please

**Pull request merged**